compiler/src/test/java/com/squareup/anvil/compiler/InterfaceMergerRepeatableTest.kt

```diff
@@ -0,0 +1,189 @@
+package com.squareup.anvil.compiler
+
+import com.google.common.truth.Truth.assertThat
+import com.squareup.anvil.annotations.MergeComponent
+import com.squareup.anvil.annotations.MergeSubcomponent
+import com.squareup.anvil.annotations.compat.MergeInterfaces
+import com.squareup.anvil.compiler.internal.testing.extends
+import org.junit.Assume.assumeTrue
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+import org.junit.runners.Parameterized.Parameters
+import kotlin.reflect.KClass
+
+@RunWith(Parameterized::class)
+class InterfaceMergerRepeatableTest(
+  private val annotationClass: KClass<*>
+) {
+
+  private val annotation = "@${annotationClass.simpleName}"
+  private val import = "import ${annotationClass.java.canonicalName}"
+
+  companion object {
+    @Parameters(name = "{0}")
+    @JvmStatic fun annotationClasses(): Collection<Any> {
+      return buildList {
+        add(MergeComponent::class)
+        if (isFullTestRun()) {
+          add(MergeSubcomponent::class)
+          add(MergeInterfaces::class)
+        }
+      }
+    }
+  }
+
+  @Test fun `duplicate scopes are an error`() {
+    assumeIrBackend()
+
+    compile(
+      """
+      package com.squareup.test
+      
+      $import
+      
+      $annotation(Any::class)
+      $annotation(Any::class)
+      interface ComponentInterface
+      """
+    ) {
+      assertThat(exitCode).isError()
+      assertThat(messages).contains(
+        "com.squareup.test.ComponentInterface merges multiple times to the same scope: [Any]. " +
+          "Merging multiple times to the same scope is forbidden and all scopes must be distinct."
+      )
+    }
+  }
+
+  @Test fun `different kind of merge annotations are forbidden`() {
```

Comment:
Given that the class/suite is parameterized, will this test still run for each annotation even though we're not using the parameter/`annotationClass` field in this test?

Reply:
Yes, it will. That's why the test uses `assumeTrue(annotationClass == MergeComponent::class)` to make it run only once. For all other annotations the test will be skipped.